### PR TITLE
Fix heredoc

### PR DESCRIPTION
**Pull request by rhysd**
### Code

```crystal
s = <<-HERE
Hello, mom! I am HERE.
HER dress is beautiful.
HE is OK.
  HERESY
aaaaaa
HERE
puts s
```

### Output

```
Hello, mom! I am HERE.HER dress is beautiful.HE is OK.  HERESY
aaaaaa
```

Newlines were discarded wrongly when line started with the same character as heredoc identifier.  Tests were also wrong.
I fixed it by adding missing newline to token value.  And I also added a test for heredoc on parse phase.

### CODE REVIEW

File: src/compiler/crystal/syntax/lexer.cr [hunk 1]
```diff
@@ -1703,6 +1703,10 @@ module Crystal
               @reader.pos = old_pos
               @column_number = old_column
               next_string_token delimiter_state
+              value = @token.value
+              if value.is_a? String
```
Comment (asterite) [inline src/compiler/crystal/syntax/lexer.cr]: Is there a way for this not to be true? I think maybe we can do `@token.value = '\n' + @token.value.to_s`.

**Comment by asterite** [comment]
Wow, I don't know how we missed that :blush: 

I added a small comment, the fix looks good :-)

File: src/compiler/crystal/syntax/lexer.cr (hunk 1 above)
Comment (rhysd) [inline src/compiler/crystal/syntax/lexer.cr]: Looks better! I'll fix it and squash.  Thanks.

**Comment by rhysd** [comment]
@asterite 

I fixed it and confirmed all tests passed!

**Comment by asterite** [comment]
Thank you!

**Pull request merged**